cmd/kube-scheduler/app/server.go

```diff
@@ -126,18 +126,30 @@ func runCommand(cmd *cobra.Command, args []string, opts *options.Options, regist
 	if err != nil {
 		return err
 	}
-	if cc == nil {
-		return nil
-	}
 
-	return Run(ctx, cc, sched)
+	return Run(ctx, cc, opts, sched)
 }
 
 // Run executes the scheduler based on the given configuration. It only returns on error or when context is done.
-func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *scheduler.Scheduler) error {
+func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, opts *options.Options, sched *scheduler.Scheduler) error {
+	if len(opts.WriteConfigTo) > 0 {
```

Comment:
Sorry for the back-and-forth. I think we can have this directly in `runCommand` instead, right before calling Run. Otherwise, there is no other reason to pass both `cc` and `opts` to run. And we don't need to WriteConfigTo in unit tests or serving tests.

Reply:
done